bucket_map/src/bucket_storage.rs

```diff
@@ -41,13 +41,10 @@ impl Header {
         Ok(0)
             == self
                 .lock
-                .compare_exchange(0, uid, Ordering::Relaxed, Ordering::Relaxed)
+                .compare_exchange(0, uid, Ordering::Acquire, Ordering::Relaxed)
     }
-    fn unlock(&self, uid: u64) -> bool {
-        Ok(uid)
-            == self
-                .lock
-                .compare_exchange(uid, 0, Ordering::Relaxed, Ordering::Relaxed)
+    fn unlock(&self) -> u64 {
+        self.lock.swap(0, Ordering::Release)
```

Comment:
1. Always unlock, unconditionally
2. Store using `Release`
3. Return the old value so we can still check for errors